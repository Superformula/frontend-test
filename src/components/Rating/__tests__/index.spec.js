import React from 'react';
import { shallow, mount } from 'enzyme';
import { Rating } from '../';

describe('Rating', () => {
  it('should render without failing', () => {
    const wrapper = shallow(<Rating />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a custom number of stars', () => {
    const wrapper = shallow(<Rating max={2} rating={1} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a half stars', () => {
    const wrapper = shallow(<Rating max={2} rating={1.5} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a small rating', () => {
    const wrapper = shallow(<Rating $xs />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a large rating', () => {
    const wrapper = shallow(<Rating $lg />);
    expect(wrapper).toMatchSnapshot();
  });
});
