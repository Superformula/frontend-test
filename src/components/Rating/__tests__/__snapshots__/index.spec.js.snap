// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating should render a custom number of stars 1`] = `
<Wrapper>
  <Star
    fill={true}
  />
  <Star
    fill={false}
  />
</Wrapper>
`;

exports[`Rating should render without failing 1`] = `
<Wrapper>
  <Star
    fill={false}
  />
  <Star
    fill={false}
  />
  <Star
    fill={false}
  />
  <Star
    fill={false}
  />
  <Star
    fill={false}
  />
</Wrapper>
`;
